core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -745,24 +786,27 @@ public void run() {
 
   @Override
   public void resetConnectBackoff() {
-    channelExecutor.executeLater(
-        new Runnable() {
-          @Override
-          public void run() {
-            if (shutdown.get()) {
-              return;
-            }
-            if (nameResolverStarted) {
-              nameResolver.refresh();
-            }
-            for (InternalSubchannel subchannel : subchannels) {
-              subchannel.resetConnectBackoff();
-            }
-            for (InternalSubchannel oobChannel : oobChannels) {
-              oobChannel.resetConnectBackoff();
-            }
-          }
-        }).drain();
+    channelExecutor
+        .executeLater(
+            new Runnable() {
+              @Override
+              public void run() {
+                if (shutdown.get()) {
+                  return;
+                }
+                if (nameResolverBackoffFuture != null) {
+                  cancelNameResolverBackoff();
```

Comment:
For clarity, could you add an `assert nameResolverStarted;`? That also helps during auditing.

Reply:
Done